### PR TITLE
[Procheta | Japneet] Fix. select api flow

### PR DESCRIPTION
**Pull request by japneetkaurbhatia**

### CODE REVIEW

File: local.env [hunk 1]
```diff
@@ -4,4 +4,5 @@ SELLER_APP_URL=http://localhost:3010
 GATEWAY_URL=http://localhost:1010
 PRIVATE_KEY=sUUXM0UOgF1WuBgbG1ILQWrqtWZTeN5XMVlMX2DsqqNNRg3uO0a/htwq5aLASya+mWtPeZPwEykw4NBT1LHosQ==
 PUBLIC_KEY=TUYN7jtGv4bcKuWiwEsmvplrT3mT8BMpMODQU9Sx6LE=
-REGISTRY_URL=http://localhost:1010
\ No newline at end of file
+REGISTRY_URL=http://localhost:1010
+UKID_BAP = 111-222-301
```
Comment (hariharanweb) [inline local.env]: Dont hard code someone elses ukId here

File: src/controllers/SelectController.js [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+import LookUpService from '../services/LookUpService';
 
-const select = (req, res) => {
+const select = async (req, res) => {
   const logger = log4js.getLogger('SelectController');
   logger.debug(`Select called with ${JSON.stringify(req.body)}`);
-
-  SelectService.select(req.body);
-  GenericResponse.sendAcknowledgement(res);
+  const publicKey = await LookUpService.getPublicKeyWithUkId(process.env.UKID_BAP);
```
Comment (hariharanweb) [inline src/controllers/SelectController.js]: Look at context to get the bap_id

Comment (hariharanweb) [reply src/controllers/SelectController.js]: Then get public key of bap_id

File: local.env (hunk 1 above)
Comment (japneetkaurbhatia) [reply local.env]: Roling back the changes

File: src/controllers/SelectController.js (hunk 2 above)
Comment (japneetkaurbhatia) [reply src/controllers/SelectController.js]: Roling back the changes

File: src/controllers/SelectController.js [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
+import LookUpService from '../services/LookUpService';
 
-const select = (req, res) => {
+const select = async (req, res) => {
   const logger = log4js.getLogger('SelectController');
   logger.debug(`Select called with ${JSON.stringify(req.body)}`);
-
-  SelectService.select(req.body);
-  GenericResponse.sendAcknowledgement(res);
+  const publicKey = await LookUpService.getPublicKeyWithUkId(req.body.context.ukId));
```
Comment (hariharanweb) [inline src/controllers/SelectController.js]: use bap_id from context and not ukId

**Pull request merged**